test/mpi/test_mpi.cc

```diff
@@ -523,6 +523,9 @@ std::vector<ucc_test_mpi_result_t> UccTestMpi::exec_tests(
             }
         } while (num_done != tcs.size());
         for (auto tc: tcs) {
+            if (tc->args.flags & UCC_COLL_ARGS_FLAG_MEM_MAPPED_BUFFERS) {
+                MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
confirm if this is needed or if just need an extra barrier before the test and check

Reply:
can confirm that only need the barrier. not the extra buffers. However, the barrier should be prior to collective start.